### PR TITLE
feat:added timeline

### PR DESCRIPTION
**Pull request by Seventy7dot2**
### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your default branch!
- [x] Make sure you are making a pull request against the **default branch** (left side). Also you should start *your branch* off *default branch*.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.
- [x] I have added necessary documentation (if appropriate)

### Description
Please describe your pull request.

❤️Thank you!

### Post merge checklist
- [x] Follow steps from the [guidelines for contributing](../CONTRIBUTING.md) to this repository.
- [ ] If you are a new contributor, ping in the thread and one of the maintainers will add you to all-contributors list.

### CODE REVIEW

File: src/components/timeline/timeline.css
Comment (Lucif3r-in) [inline src/components/timeline/timeline.css:1]: Do not use css file use styled components + twin macro for styles.

File: src/components/timeline/Timeline.jsx
Comment (Lucif3r-in) [inline src/components/timeline/Timeline.jsx:1]: Map the piece of code as it will be used multiple times.

File: .DS_Store
Comment (Lucif3r-in) [inline .DS_Store:1]: What is use of these files?

File: src/.DS_Store
Comment (Lucif3r-in) [inline src/.DS_Store:1]: Remove this

File: src/components/.DS_Store
Comment (Lucif3r-in) [inline src/components/.DS_Store:1]: Remove this

**Review by Lucif3r-in — CHANGES_REQUESTED** [review]

**Comment by SirSimon162** [comment]
@Seventy7dot2 
- Use styled components and twin macro.
- Make the mobile view as per the design.

**Comment by Seventy7dot2** [comment]
@Lucif3r-in @SirSimon162 What will be the breaking point for mobile view?

**Comment by Lucif3r-in** [comment]
> @Lucif3r-in @SirSimon162 What will be the breaking point for mobile view?

You have to keep two breakpoints, for tablets 768px and for mobile- 480px

File: src/components/timeline/Extensionboth.jsx [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+    color: ${(props) => props.color};
+    font-family: Prompt;
+    font-size: 32px;
+    font-style: normal;
+    font-weight: 600;
+    line-height: 48px;
+    text-transform: capitalize;
+    margin: 0;
+  }
+}
```
Comment (Lucif3r-in) [inline src/components/timeline/Extensionboth.jsx:45]: Use typography

File: src/components/timeline/Extensionboth.jsx [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+p{
+  color: var(--neutral-5, #A1A1A1);
```
Comment (Lucif3r-in) [inline src/components/timeline/Extensionboth.jsx:47]: use color from tailwindconfig.js

File: src/components/timeline/Extensionboth.jsx [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+    font-family: Prompt;
+    font-size: 16px;
+    font-style: normal;
+    font-weight: 400;
+    line-height: 20px;
+    margin: 0;
```
Comment (Lucif3r-in) [inline src/components/timeline/Extensionboth.jsx:53]: use typography

File: src/components/timeline/Extensionboth.jsx [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+      align-self: stretch;
+    }
+  }
+    p {
+      font-size: 10px;
+      font-style: normal;
+      font-weight: 300;
+      line-height: 18px;
+    }  
+}
```
Comment (Lucif3r-in) [inline src/components/timeline/Extensionboth.jsx:83]: Use typography as per figma

File: src/components/timeline/Extensionboth.jsx [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+  color: #FFF;
+  /* Desktop/Body/B1/Regular */
+  font-family: Prompt;
+  font-size: 18px;
+  font-style: normal;
+  font-weight: 300;
+  line-height: 28px;
+  /* 155.556% */
+  margin: 0;
+}
```
Comment (Lucif3r-in) [inline src/components/timeline/Extensionboth.jsx:181]: Use typography

File: src/components/timeline/Extensionboth.jsx [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
+      </Mobile>
+    </>
+  )
+};
+
+Extensionboth.defaultProps = {
+  boxcolor: "#FEE27F",
+  wheelcolor: "#C4C4C4",
+  ellipsecolor: "#FF65CD"
+}
```
Comment (Lucif3r-in) [inline src/components/timeline/Extensionboth.jsx:400]: Use colors from tailwindconfig

File: src/components/timeline/timeline.css
Comment (Lucif3r-in) [inline src/components/timeline/timeline.css:1]: Delete this file

File: src/pages/playground.js [hunk 7]
```diff
@@ -1,20 +1,18 @@
...
+import Timeline from '../components/timeline/Timeline'
 import { Body1, Body2, Heading1, Heading2, Heading3, Heading4, NavText, ButtonMeta, ButtonMetaLarge } from '../components';
 
 export default function Playground() {
+  let event={
+    date:15,
+    month:"October",
+    title:"Registration Opens",
+    date_end:"15 dec"
+    }
```
Comment (Lucif3r-in) [inline src/pages/playground.js:12]: Keep the contents at config/content/Timeline.js outside src

**Review by Lucif3r-in — CHANGES_REQUESTED** [review]

File: src/components/timeline/Timeline.jsx
Comment (Seventy7dot2) [reply src/components/timeline/Timeline.jsx:1]: Where to create the content config file?

Comment (Lucif3r-in) [reply src/components/timeline/Timeline.jsx:1]: In the root, there will be `config`, create a `content` folder in it then create `Timeline.js` in there

**Comment by Lucif3r-in** [comment]
@Seventy7dot2 
- When you give some padding to the section it overflows. Work on it
- After 1258px it fails in responsiveness and make margin-auto so that it is always centered

**Comment by Seventy7dot2** [comment]
> @Seventy7dot2
> 
> * When you give some padding to the section it overflows. Work on it
> * After 1258px it fails in responsiveness and make margin-auto so that it is always centered

I have made some changes. Can you check it now? and if the problem persists please attach the screenshot

**Comment by Lucif3r-in** [comment]
Please follow commit message standards

File: src/components/timeline/Extensionboth.jsx (hunk 1 above)
Comment (Lucif3r-in) [reply src/components/timeline/Extensionboth.jsx:45]: Resolve these requested changes

File: src/components/timeline/Extensionboth.jsx [hunk 8]
```diff
@@ -0,0 +1,402 @@
...
+
+const Container = styled.div`
+    display: grid;
+    grid-template-columns: 1fr 201px 134px 243px 1fr;
+    grid-template-rows: 248px 135px;
+    width: 1240px;
+    margin: auto;
+    color:#FFF;
+    @media (max-width: 640px) {
+  }
```
Comment (Lucif3r-in) [inline src/components/timeline/Extensionboth.jsx:13]: remove unused code

File: src/components/timeline/Extensionboth.jsx (hunk 2 above)
Comment (Lucif3r-in) [reply src/components/timeline/Extensionboth.jsx:41]: Also use typorgaphy

File: src/components/timeline/Extensionboth.jsx [hunk 9]
```diff
@@ -0,0 +1,402 @@
...
+      align-self: stretch;
+    }
+  }
+    p {
+      font-size: 10px;
+      font-style: normal;
+      font-weight: 300;
+      line-height: 18px;
+    }  
+}
```
Comment (Lucif3r-in) [inline src/components/timeline/Extensionboth.jsx:71]: Use typography

File: src/components/timeline/Extensionboth.jsx [hunk 10]
```diff
@@ -0,0 +1,402 @@
...
...6 236.6 1C236.6 1.22091 236.779 1.4 237 1.4V0.6ZM245 1.4C245.221 1.4 245.4 1.22091 245.4 1C245.4 0.779086 245.221 0.6 245 0.6V1.4ZM253 0.6C252.779 0.6 252.6 0.779086 252.6 1C252.6 1.22091 252.779 1.4 253 1.4V0.6ZM261 1.4C261.221 1.4 261.4 1.22091 261.4 1C261.4 0.779086 261.221 0.6 261 0.6V1.4ZM269 0.6C268.779 0.6 268.6 0.779086 268.6 1C268.6 1.22091 268.779 1.4 269 1.4V0.6ZM277 1.4C277.221 1.4 277.4 1.22091 277.4 1C277.4 0.779086 277.221 0.6 277 0.6V1.4ZM285 0.6C284.779 0.6 284.6 0.779086 284.6 1C284.6 1.22091 284.779 1.4 285 1.4V0.6ZM293 1.4C293.221 1.4 293.4 1.22091 293.4 1C293.4 0.779086 293.221 0.6 293 0.6V1.4ZM301 0.6C300.779 0.6 300.6 0.779086 300.6 1C300.6 1.22091 300.779 1.4 301 1.4V0.6ZM1 1.4H5V0.6H1V1.4ZM13 1.4H21V0.6H13V1.4ZM29 1.4H37V0.6H29V1.4ZM45 1.4H53V0.6H45V1.4ZM61 1.4H69V0.6H61V1.4ZM77 1.4H85V0.6H77V1.4ZM93 1.4H101V0.6H93V1.4ZM109 1.4H117V0.6H109V1.4ZM125 1.4H133V0.6H125V1.4ZM141 1.4H149V0.6H141V1.4ZM157 1.4H165V0.6H157V1.4ZM173 1.4H181V0.6H173V1.4ZM189 1.4H197V0.6H189V1.4ZM205 1.4H213V0.6H205V1.4ZM221 1.4H229V0.6H221V1.4ZM237 1.4H245V0.6H237V1.4ZM253 1.4H261V0.6H253V1.4ZM269 1.4H277V0.6H269V1.4ZM285 1.4H293V0.6H285V1.4ZM301 1.4H305V0.6H301V1.4Z" fill="#C4C4C4" />
```
Comment (Lucif3r-in) [inline src/components/timeline/Extensionboth.jsx:79]: Also keep this in content

File: src/components/timeline/Extensionboth.jsx [hunk 11]
```diff
@@ -0,0 +1,402 @@
...
+      <svg xmlns="http://www.w3.org/2000/svg" width="202" height="2" viewBox="0 0 202 2" fill="none">
+        <path
+          d="M1 0.5C0.723858 0.5 0.5 0.723858 0.5 1C0.5 1.27614 0.723858 1.5 1 1.5L1 0.5ZM1 1.5L202 1.49998L202 0.499982L1 0.5L1 1.5Z"
+          fill={props.color} />
+      </svg>
+    </Rightlinewrapper>
+  )
+}
+
+
```
Comment (Lucif3r-in) [inline src/components/timeline/Extensionboth.jsx:137]: remove unnecessary spaces

File: src/components/timeline/Extensionboth.jsx [hunk 12]
```diff
@@ -0,0 +1,402 @@
...
+    align-items: center;
+    .line1 {
+      width: 4px;
+      height: 197px;
+      background: #FED954;
+      transform: translateY(-27px);
+  }
+  @media(max-width:640px){
+      /* display:none; */
+  }
```
Comment (Lucif3r-in) [inline src/components/timeline/Extensionboth.jsx:150]: Remove unused code

File: src/components/timeline/Extensionboth.jsx [hunk 13]
```diff
@@ -0,0 +1,402 @@
...
+  return (
+    <Wheelwrapper>
+      <svg xmlns="http://www.w3.org/2000/svg" width="134" height="134" viewBox="0 0 134 134" fill="none">
+        <circle cx="67" cy="67" r="65" stroke={props.color} strokeWidth="4" />
+        <circle cx="67.0002" cy="67" r="48.8524" stroke={props.color} strokeWidth="1.2"
+          strokeLinecap="round" strokeLinejoin="round" strokeDasharray="10 10" />
+      </svg>
+      <div className="content">
+        <p id="date">{props.node.date}</p>
+        <Body1 id="month">{props.node.month}</Body1>
```
Comment (Lucif3r-in) [inline src/components/timeline/Extensionboth.jsx:226]: Use typography

File: src/components/timeline/Extensionboth.jsx [hunk 14]
```diff
@@ -0,0 +1,402 @@
...
+      </div>
+    </Wheelwrapper>
+  )
+}
+
+
```
Comment (Lucif3r-in) [inline src/components/timeline/Extensionboth.jsx:232]: remove unnecessary space

File: src/components/timeline/Extensionright.jsx [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+import styled from 'styled-components'
+
+const Rightcontainer = styled.div`
+display: grid;
+grid-template-columns: 352px 201px 134px 201px 1fr;
+grid-template-rows: 141px 103px;
+width: 1240px;
+margin: auto;
+color:#FFF;
+`
```
Comment (Lucif3r-in) [inline src/components/timeline/Extensionright.jsx:12]: Use ESLint and prettier for maintaining code structure.

File: src/components/timeline/Extensionboth.jsx (hunk 10 above)
Comment (Seventy7dot2) [reply src/components/timeline/Extensionboth.jsx]: It was hard to handle such so just going on with img that doesn't affect anything

File: src/components/timeline/Extensionboth.jsx [hunk 16]
```diff
@@ -0,0 +1,218 @@
...
+export const Box = ({ node }) => (
+  <BoxWrapper color={node.boxcolor}>
+    <div>
+      <Heading3 semibold>{node.title}</Heading3>
+      <img
+        src='https://res.cloudinary.com/dgjzygzgx/image/upload/v1698431749/Line_3_o1ztvt.svg'
+        alt='line'
+      />
+    </div>
```
Comment (Lucif3r-in) [inline src/components/timeline/Extensionboth.jsx:29]: Why is there an extra div here? Move the links to content folder.

File: src/components/timeline/Extensionright.jsx [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+
+export const VerticalLine = () => (
+  <VerticalWrapper>
+    <div />
+  </VerticalWrapper>
+);
+export default function ExtensionRight({ position, wheelcolor, node, boxcolor }) {
+  return (
+    <>
+      <Desktop>
```
Comment (Lucif3r-in) [inline src/components/timeline/Extensionright.jsx:13]: Change to `DesktopView` and `MobileView` respectively

**Review by Lucif3r-in — CHANGES_REQUESTED** [review]

File: src/components/timeline/Extensionboth.jsx (hunk 16 above)
Comment (Seventy7dot2) [reply src/components/timeline/Extensionboth.jsx:29]: The div is the wrapper for H3 and img. have used it for alignment

File: src/components/timeline/Extensionright.jsx (hunk 17 above)
Comment (Seventy7dot2) [reply src/components/timeline/Extensionright.jsx]: Done

File: src/components/timeline/Extensionboth.jsx (hunk 16 above)
Comment (Lucif3r-in) [reply src/components/timeline/Extensionboth.jsx:29]: Check if that can be done with boxwrapper

Comment (Seventy7dot2) [reply src/components/timeline/Extensionboth.jsx]: Yes that was possible. Done!

**Comment by Lucif3r-in** [comment]
LGTM🚀

**Pull request merged**